scrunch/datasets.py

```diff
@@ -1441,12 +1481,34 @@ class Variable(object):
 
     CATEGORICAL_TYPES = {'categorical', 'multiple_response', 'categorical_array'}
 
-    def __init__(self, resource, dataset_resource):
-        self.resource = resource
-        self.url = self.resource.self
-        self.dataset = Dataset(dataset_resource)
+    def __init__(self, var_shoji, dataset):
+        self.shoji_tuple = var_shoji
+        self.is_instance = False
+        self._resource = None
+        self.url = None
+        # pass an Entity class to save time
+        if isinstance(dataset, pycrunch.shoji.Entity):
+            self.dataset = dataset
+        else:
+            self.dataset = Dataset(dataset)
```

Comment:
yes, it will equally work with self.dataset = dataset.entity instead of Dataset(dataset)